modules/tensor_mechanics/src/interfacekernels/CZMInterfaceKernel.C

```diff
@@ -0,0 +1,222 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CZMInterfaceKernel.h"
+
+registerMooseObject("TensorMechanicsApp", CZMInterfaceKernel);
+
+template <>
+InputParameters
+validParams<CZMInterfaceKernel>()
+{
+  InputParameters params = validParams<InterfaceKernel>();
+  params.addRequiredParam<unsigned int>("disp_index",
+                                        "the component of the "
+                                        "displacement vector this kernel is working on:"
+                                        " disp_index == 0, ==> X"
+                                        " disp_index == 1, ==> Y"
+                                        " disp_index == 2, ==> Z");
+  params.addCoupledVar("disp_1",
+                       "Name of the variable representing the 2nd "
+                       "displacement to couple on the master side. "
+                       "If disp_index == 0, then disp_1 = disp_y "
+                       "If disp_index == 1, then disp_1 = disp_x "
+                       "If disp_index == 2, then disp_1 = disp_x");
+  params.addCoupledVar("disp_1_neighbor",
+                       "Name of the variable representing "
+                       "the 2nd displacement to couple on the slave side. "
+                       "If disp_index == 0, disp_1_neighbor = disp_y_neighbor "
+                       "If disp_index == 1, disp_1_neighbor = disp_x_neighbor "
+                       "If disp_index == 2, disp_1_neighbor = disp_x_neighbor");
+  params.addCoupledVar("disp_2",
+                       "Name of the variable representing the 3rd "
+                       "displacement to couple on the master side. "
+                       "If disp_index == 0, then disp_2 = disp_z "
+                       "If disp_index == 1, then disp_2 = disp_z "
+                       "If disp_index == 2, then disp_2 = disp_y ");
+  params.addCoupledVar("disp_2_neighbor",
+                       "Name of the variable representing "
+                       "the 3rd displacement to couple on the slave side. "
+                       "If disp_index == 0, disp_2_neighbor = disp_z_neighbor "
+                       "If disp_index == 1, disp_2_neighbor = disp_z_neighbor "
+                       "If disp_index == 2, disp_2_neighbor = disp_y_neighbor");
+  params.addParam<std::string>(
+      "residual",
+      "traction",
+      "The name of the material property representing the residual coefficients");
+  params.addParam<std::string>(
+      "jacobian",
+      "traction_spatial_derivatives",
+      "The name of the  material property representing the jacobian coefficients");
```

Comment:
I would like to change residual -> residual and jacobian-> traction_derivatives

Reply:
this has been changed